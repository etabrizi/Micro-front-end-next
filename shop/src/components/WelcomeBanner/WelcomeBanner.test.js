import * as React from 'react'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import { WelcomeBanner } from '.'

const BaseProps = {
  title: 'Title',
  intro: 'Intro copy',
  children: 'Children content'
}

afterEach(cleanup)

describe('Welcome banner', () => {
  it('renders without crashing', () => {
    const { asFragment } = render(<WelcomeBanner {...BaseProps} />)
    expect(asFragment()).toMatchSnapshot()
  })
  it('has a title', () => {
    const { getByText } = render(<WelcomeBanner {...BaseProps} />)
    expect(getByText('Title')).toBeInTheDocument()
  })
  it('has an intro', () => {
    const { getByText } = render(<WelcomeBanner {...BaseProps} />)
    expect(getByText('Intro copy')).toBeInTheDocument()
  })
  it('has some child content', () => {
    const { getByText } = render(<WelcomeBanner {...BaseProps} />)
    expect(getByText('Children content')).toBeInTheDocument()
  })
})
