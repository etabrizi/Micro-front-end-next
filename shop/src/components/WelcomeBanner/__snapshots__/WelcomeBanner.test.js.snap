// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome banner renders without crashing 1`] = `
<DocumentFragment>
  <section
    class="welcome-banner"
  >
    <div
      class="welcome-banner__container"
    >
      <h1
        class="welcome-banner__title"
      >
        Title
      </h1>
      <p
        class="welcome-banner__intro"
      >
        Intro copy
      </p>
      <div
        class="welcome-banner__content"
      >
        Children content
      </div>
    </div>
  </section>
</DocumentFragment>
`;
